apps/protailwind/src/templates/workshop-template.tsx

```diff
@@ -35,28 +57,116 @@ const WorkshopTemplate: React.FC<{
+      <Header workshop={workshop} purchased={hasPurchased} />
+      <main
+        data-workshop={workshop.slug}
+        className="relative z-10 flex flex-col gap-8 lg:flex-row lg:items-start"
+      >
+        <div className="max-w-xl">
+          <article className="prose w-full max-w-none pb-10 text-gray-900 lg:max-w-xl">
+            <PortableText value={body} components={PortableTextComponents} />
+          </article>
+          <WorkshopSectionNavigator
+            purchased={hasPurchased}
+            workshop={workshop}
+          />
+        </div>
+        <BuyWorkshop
+          product={product}
+          workshop={workshop}
+          {...propsForCommerce}
+        />
       </main>
     </Layout>
   )
 }
 
 export default WorkshopTemplate
 
-const Header: React.FC<{workshop: SanityDocument}> = ({workshop}) => {
+const BuyWorkshop: React.FC<
+  CommerceProps & {product: SanityProduct; workshop: SanityDocument}
+> = ({
+  product,
+  workshop,
+  couponFromCode,
+  purchases = [],
+  userId,
+  products,
+  couponIdFromCoupon,
+  defaultCoupon,
+  allowPurchase,
+}) => {
+  const {redeemableCoupon, RedeemDialogForCoupon, validCoupon} =
+    useCoupon(couponFromCode)
+  const couponId =
+    couponIdFromCoupon || (validCoupon ? couponFromCode?.id : undefined)
+  const purchasedProductIds = purchases.map((purchase) => purchase.productId)
+  const hasPurchased =
+    purchasedProductIds && purchasedProductIds.includes(product.productId)
+  const firstLesson = workshop.sections[0].lessons[0]
+  const thumbnail = `https://protailwind.com/api/video-thumb?videoResourceId=${firstLesson?.videoResourceId}`
+  // const thumbnail = `${getBaseUrl()}/api/video-thumb?videoResourceId=${firstLesson?.videoResourceId}`
+
+  return isSellingLive ? (
+    <div className="mx-auto w-full max-w-sm overflow-hidden rounded-lg border border-gray-200/40 bg-white shadow-2xl shadow-gray-400/20">
+      {!hasPurchased && (
+        <Link
+          href={{
+            pathname: '/workshops/[module]/[section]/[lesson]',
+            query: {
+              section: workshop.sections[0].slug,
+              lesson: firstLesson.slug,
+              module: workshop.slug.current,
+            },
+          }}
+        >
+          <a className="group relative flex aspect-video h-full w-full items-center justify-center bg-black">
+            <Image
+              src={thumbnail}
+              layout="fill"
+              alt=""
+              aria-hidden="true"
+              objectFit="cover"
+              className="opacity-80"
+            />
+            <PlayIcon
+              className="absolute h-10 w-10 text-white transition group-hover:scale-105"
+              aria-hidden="true"
+            />
+            <div className="absolute left-0 bottom-0 flex w-full items-center justify-center bg-gradient-to-t from-black/80 to-transparent pb-3.5 pt-8 text-sm font-medium text-white">
+              Preview this workshop
```

Comment:
ux consideration: this could just open the video(s) in a modal rather than linking to a different page to avoid loosing context.